### PR TITLE
Add some basic benchmarks.

### PR DESCRIPTION
**Pull request by dfish3r**
Intended to provide some baseline benchmarks to measure throughput performance.
Could provide meaningful, quantitative data for knowing whether a change improved performance.

### CODE REVIEW

File: benchmark/pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>ldaptive-benchmark</artifactId>
```
Comment (serac) [inline benchmark/pom.xml]: Did you generate this artifact/module using the Maven archtype as recommended in the docs, https://openjdk.java.net/projects/code-tools/jmh/?

Comment (dfish3r) [reply benchmark/pom.xml]: I believe I did, but then immediately hacked it up to look like the existing pom files. Did I remove something important?

**Comment by serac** [comment]
I'm not familiar with JMH, but a cursory read of https://www.oracle.com/technical-resources/articles/java/architect-benchmarking.html makes it clear that it's the only way to produce meaningful benchmarks on Java code whose throughput is not governed by IO wait, though it's probably preferable in those cases as well.

File: benchmark/pom.xml (hunk 1 above)
Comment (serac) [reply benchmark/pom.xml]: Not that I can see. I was mostly just curious.

**Pull request merged**